Fails to build on the i386 architecture: non-constant-expression cannot be narrowed from type 'xgboost::bst_ulong'

```
/wrkdirs/usr/ports/misc/xgboost/work/xgboost-0.90-291-g0c745527/src/c_api/c_api.cc:314:52: error: non-constant-expression cannot be narrowed from type 'xgboost::bst_ulong' (aka 'unsigned long long') to 'xgboost::common::Span::index_type' (aka 'unsigned int') in initializer list [-Wc++11-narrowing]
  data::DMatrixSliceAdapter adapter(dmat, {idxset, len});
                                                   ^~~
/wrkdirs/usr/ports/misc/xgboost/work/xgboost-0.90-291-g0c745527/src/c_api/c_api.cc:314:52: note: insert an explicit cast to silence this issue
  data::DMatrixSliceAdapter adapter(dmat, {idxset, len});
                                                   ^~~
                                                   static_cast<index_type>( )
1 error generated.
```


See the log: http://beefy17.nyi.freebsd.org/data/head-i386-default/p524616_s357279/logs/xgboost-0.90.291.log (IPv6 URL)

--- a/src/c_api/c_api.cc
+++ b/src/c_api/c_api.cc
@@ -311,7 +311,7 @@ XGB_DLL int XGDMatrixSliceDMatrixEx(DMatrixHandle handle,
   DMatrix* dmat = static_cast<std::shared_ptr<DMatrix>*>(handle)->get();
   CHECK(dynamic_cast<data::SimpleDMatrix*>(dmat))
       << "Slice only supported for SimpleDMatrix currently.";
-  data::DMatrixSliceAdapter adapter(dmat, {idxset, len});
+  data::DMatrixSliceAdapter adapter(dmat, {idxset, static_cast<size_t>(len)});
   *out = new std::shared_ptr<DMatrix>(
       DMatrix::Create(&adapter, std::numeric_limits<float>::quiet_NaN(), 1));
   API_END();

--- a/src/learner.cc
+++ b/src/learner.cc
@@ -670,9 +670,10 @@ class LearnerImpl : public Learner {
   for more details about differences between saving model and serializing.
 
 )doc";
-      int64_t json_offset {-1};
-      CHECK_EQ(fp.Read(&json_offset, sizeof(json_offset)), sizeof(json_offset));
-      CHECK_GT(json_offset, 0);
+      int64_t sz {-1};
+      CHECK_EQ(fp.Read(&sz, sizeof(sz)), sizeof(sz));
+      CHECK_GT(sz, 0);
+      size_t json_offset = static_cast<size_t>(sz);
       std::string buffer;
       common::FixedSizeStream{&fp}.Take(&buffer);
 

